modular_skyrat/modules/armaments/code/armament_station.dm

```diff
@@ -0,0 +1,209 @@
+					if(armament_entry)
+						break
+				if(armament_entry)
+					break
+			if(!armament_entry)
+				return
+			if(products && !(armament_entry.type in products))
+				return
+			select_armament(usr, armament_entry)
+		if("eject_card")
+			eject_card(usr)
+
+/obj/machinery/armament_station/proc/eject_card(mob/user)
+	if(!inserted_card)
+		to_chat(user, span_warning("No card inserted!"))
+		return
+	inserted_card.forceMove(drop_location())
+	user.put_in_hands(inserted_card)
+	inserted_card = null
+	to_chat(user, span_notice("Card ejected!"))
+	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 70)
+
+/obj/machinery/armament_station/proc/select_armament(mob/user, datum/armament_entry/armament_entry)
+	if(!inserted_card)
+		to_chat(user, span_warning("No card inserted!"))
+		return
+	if(used_categories[armament_entry.category] >= GLOB.armament_entries[armament_entry.category][CATEGORY_LIMIT])
+		to_chat(user, span_warning("Category limit reached!"))
+		return
+	if(purchased_items[armament_entry] >= armament_entry.max_purchase)
+		to_chat(user, span_warning("Item limit reached!"))
+		return
+	if(!ishuman(user))
+		return
+	if(!inserted_card.use_points(armament_entry.cost))
+		to_chat(user, span_warning("Not enough points!"))
+		return
+
+	var/mob/living/carbon/human/human_to_equip = user
+
+	var/obj/item/new_item = new armament_entry.item_type(drop_location())
+
+	used_categories[armament_entry.category]++
+	purchased_items[armament_entry]++
+
+	playsound(src, 'sound/machines/machine_vend.ogg', 50, TRUE, extrarange = -3)
+
+	if(armament_entry.equip_to_human(human_to_equip, new_item))
+		to_chat(user, span_notice("Equipped directly to your person."))
+		playsound(src, 'sound/items/equip/toolbelt_equip.ogg', 100)
+	armament_entry.after_equip(drop_location(), new_item)
+
+/**
+ * Armament points card
+ *
+ * To be used with the armaments vendor.
+ */
+/obj/item/armament_points_card
+	name = "armament points card"
+	desc = "A points card that can be used at an Armaments Station or Armaments Dealer."
+	icon = 'modular_skyrat/modules/armaments/icons/armaments.dmi'
+	icon_state = "armament_card"
+	/// How many points does this card have to use at the vendor?
+	var/points = 10
+
+/obj/item/armament_points_card/Initialize(mapload)
+	. = ..()
+	maptext = span_maptext("<div align='center' valign='middle' style='position:relative'>[points]</div>")
+
+/obj/item/armament_points_card/examine(mob/user)
+	. = ..()
+	. += span_notice("It has [points] points left.")
+
+/obj/item/armament_points_card/proc/use_points(points_to_use)
+	if(points_to_use > points)
+		return FALSE
+
+	points -= points_to_use
+
+	update_maptext()
+
+	return TRUE
+
+/obj/item/armament_points_card/proc/update_maptext()
+	maptext = span_maptext("<div align='center' valign='middle' style='position:relative'>[points]</div>")
+
+/obj/item/armament_points_card/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+	if(istype(attacking_item, /obj/item/armament_points_card))
```

Comment:
early returrn

Reply:
This will have more function added, these are cases where an early return is not acceptable

Reply:
Ditto